File: Maximum Subarray.c
int comp (int a, int b)
{
    return a>= b ? a:b;
}

int maxSubArray(int* nums, int numsSize){
    int max = nums[0], temp = nums[0];
    for( int i =1; i < numsSize; i++)
    {
        temp = comp(temp + nums[i], nums[i]);
        max = comp(max, temp);
    }
    return max;
}

File: Maximum Subarray.py
class Solution:
    

    def maxSubArray(self, nums: List[int]) -> int:
    

        maximum = nums[0]
        temp = nums[0]
        for i  in range(1, len(nums)):
            temp = max(temp + nums[i], nums[i])
            maximum = max( maximum,temp)
            
        return maximum
            
